src/Validator/Rules/PossibleTypeExtensions.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Validator\Rules;
+
+use GraphQL\Error\Error;
+use GraphQL\Error\InvariantViolation;
+use GraphQL\Language\AST\Node;
+use GraphQL\Language\AST\NodeKind;
+use GraphQL\Language\AST\TypeDefinitionNode;
+use GraphQL\Language\VisitorOperation;
+use GraphQL\Type\Definition\EnumType;
+use GraphQL\Type\Definition\InputObjectType;
+use GraphQL\Type\Definition\InterfaceType;
+use GraphQL\Type\Definition\NamedType;
+use GraphQL\Type\Definition\ObjectType;
+use GraphQL\Type\Definition\ScalarType;
+use GraphQL\Type\Definition\Type;
+use GraphQL\Type\Definition\UnionType;
+use GraphQL\Utils\Utils;
+use GraphQL\Validator\SDLValidationContext;
+
+/**
+ * Possible type extension.
+ *
+ * A type extension is only valid if the type is defined and has the same kind.
+ */
+class PossibleTypeExtensions extends ValidationRule
+{
+    public function getSDLVisitor(SDLValidationContext $context): array
+    {
+        $schema = $context->getSchema();
+        /** @var array<string, TypeDefinitionNode&Node> $definedTypes */
+        $definedTypes = [];
+
+        foreach ($context->getDocument()->definitions as $def) {
+            if ($def instanceof TypeDefinitionNode) {
+                $definedTypes[$def->name->value] = $def;
+            }
+        }
+
+        $checkTypeExtension = static function ($node) use ($context, $schema, &$definedTypes): ?VisitorOperation {
+            $typeName = $node->name->value;
+            $defNode = $definedTypes[$typeName] ?? null;
+            $existingType = $schema !== null ? $schema->getType($typeName) : null;
```

Comment:
Ternary should always be multiline, I just documented this in `CONTRIBUTING.md`

Reply:
I honestly can't see how the multiline version will make the code more readable or maintainable in this case. Anyway, is it possible to enforce it for php-cs-fixer as was for php_codesniffer in https://github.com/webonyx/graphql-php/pull/951?

Reply:
Unfortunately not.

In this case, one advantage would be that an error thrown in `$schema->getType()` would have a more specific line number when debugging.

In general, it is a matter of consistency for me.